components/collator/src/options.rs

```diff
@@ -490,61 +561,89 @@ impl CollatorOptions {
-        self.0 &= !(CollatorOptions::CASE_FIRST_MASK | CollatorOptions::UPPER_FIRST_MASK);
+        self.0 &=
+            !(CollatorOptionsBitField::CASE_FIRST_MASK | CollatorOptionsBitField::UPPER_FIRST_MASK);
         if let Some(case_first) = case_first {
-            self.0 |= CollatorOptions::EXPLICIT_CASE_FIRST_MASK;
+            self.0 |= CollatorOptionsBitField::EXPLICIT_CASE_FIRST_MASK;
             match case_first {
                 CaseFirst::Off => {}
                 CaseFirst::LowerFirst => {
-                    self.0 |= CollatorOptions::CASE_FIRST_MASK;
+                    self.0 |= CollatorOptionsBitField::CASE_FIRST_MASK;
                 }
                 CaseFirst::UpperFirst => {
-                    self.0 |= CollatorOptions::CASE_FIRST_MASK;
-                    self.0 |= CollatorOptions::UPPER_FIRST_MASK;
+                    self.0 |= CollatorOptionsBitField::CASE_FIRST_MASK;
+                    self.0 |= CollatorOptionsBitField::UPPER_FIRST_MASK;
                 }
             }
         } else {
-            self.0 &= !CollatorOptions::EXPLICIT_CASE_FIRST_MASK;
+            self.0 &= !CollatorOptionsBitField::EXPLICIT_CASE_FIRST_MASK;
         }
     }
 
     /// Whether second level compares the last accent difference
     /// instead of the first accent difference.
     pub fn backward_second_level(&self) -> bool {
-        (self.0 & CollatorOptions::BACKWARD_SECOND_LEVEL_MASK) != 0
+        (self.0 & CollatorOptionsBitField::BACKWARD_SECOND_LEVEL_MASK) != 0
     }
 
     /// Whether second level compares the last accent difference
     /// instead of the first accent difference.
     pub fn set_backward_second_level(&mut self, backward_second_level: Option<bool>) {
-        self.0 &= !CollatorOptions::BACKWARD_SECOND_LEVEL_MASK;
+        self.0 &= !CollatorOptionsBitField::BACKWARD_SECOND_LEVEL_MASK;
         if let Some(backward_second_level) = backward_second_level {
-            self.0 |= CollatorOptions::EXPLICIT_BACKWARD_SECOND_LEVEL_MASK;
+            self.0 |= CollatorOptionsBitField::EXPLICIT_BACKWARD_SECOND_LEVEL_MASK;
             if backward_second_level {
-                self.0 |= CollatorOptions::BACKWARD_SECOND_LEVEL_MASK;
+                self.0 |= CollatorOptionsBitField::BACKWARD_SECOND_LEVEL_MASK;
             }
         } else {
-            self.0 &= !CollatorOptions::EXPLICIT_BACKWARD_SECOND_LEVEL_MASK;
+            self.0 &= !CollatorOptionsBitField::EXPLICIT_BACKWARD_SECOND_LEVEL_MASK;
+        }
+    }
+
+    pub fn set_backward_second_level_from_enum(
+        &mut self,
+        backward_second_level: Option<BackwardSecondLevel>,
+    ) {
+        match backward_second_level {
+            Some(BackwardSecondLevel::On) => {
+                self.set_backward_second_level(Some(true));
+            }
+            Some(BackwardSecondLevel::Off) => {
+                self.set_backward_second_level(Some(false));
+            }
+            _ => self.set_backward_second_level(None),
```

Comment:
Nit: Avoid the catchall match when possible. It can conceal bugs.

```suggestion
            None => self.set_backward_second_level(None),
```